ironfist/src/cpp/game/save.cpp

```diff
@@ -406,18 +447,34 @@ void game::LoadGame(char* filnam, int newGame, int a3) {
             gpGame->ResetIronfistGameState();
 
 			if (mp->script().present()) {
-				ScriptingInitFromString(mp->script().get().c_str());
+				ScriptingInitFromString(mp->script().get());
 			}
 
-			i = 0;
+			std::map<std::string, mapVariable> mapVariables;
 			for (ironfist_map::map_t::mapVariable_const_iterator it = mp->mapVariable().begin();
-			it != mp->mapVariable().end();
-				it++, i++) {
-				SetMapVariables(it->id().get().c_str(), it->value().get().c_str());
+				it != mp->mapVariable().end(); it++) {
+				mapVariable *mapVar = new mapVariable;
+				std::string mapVariableId = it->id().get();
+				MapVarType mapVariableType = StringToMapVarType(it->type());
+				mapVar->type = mapVariableType;
+				if (isTable(mapVariableType)) {
+					luaTable *lt = new luaTable;
+					ReadMapVarXML(*lt, it->table().get());
+					mapVar->tableValue = lt;
+				} else if (isStringNumBool(mapVariableType)) {
+					std::string *sV = new std::string;
+					*sV = it->value().get();
+					mapVar->singleValue = sV;
+				} else {
+					ErrorLoadingMapVariable(mapVariableId, " A map variable can only be a table, number, string or boolean.");
+				}
+				mapVariables[mapVariableId] = *mapVar;
 			}
+			WriteMapVariablesToLUA(mapVariables);
+

```

Comment:
This is where loading crashes. 
Just before reaching this line, the program calls the map variable destructor for all the map variables in the list mapVariables.
But then, right at the next line, it calls the destructor again on a destroyed object, which raises the exception
     "Exception thrown: read access violation.
      _Parent_proxy was 0xDDDDDDDD."
I found a possibly relevant piece of information about that problem: the variable "was already deleted before but was not set to NULL afterwards".